index.test-d.ts

```diff
@@ -0,0 +1,44 @@
+import {expectError, expectType} from 'tsd';
+import repoUrlFromPackage, {type PackageJson} from './index.js';
+
+declare const packageJson: PackageJson;
+
+expectType<string>(packageJson.name);
+expectType<string | {url: string} | undefined>(packageJson.repository);
+
+const {url, warnings} = repoUrlFromPackage(packageJson);
+
+expectType<string | undefined>(url);
+expectType<string[]>(warnings);
+
+repoUrlFromPackage({
+	name: 'my-package',
+	repository: 'https://github.com/user/repo.git',
+});
+
+repoUrlFromPackage({
+	name: 'my-package',
+	repository: {
+		url: 'https://github.com/user/repo.git',
+	},
+});
+
+repoUrlFromPackage({
+	name: 'my-package',
+	repository: {
+		url: 'https://example.com',
+	},
+});
+
+repoUrlFromPackage({
+	name: 'my-package',
+	repository: {
+		url: 'invalid-url',
+	},
+});
+
+repoUrlFromPackage({
+	name: 'my-package',
+});
+
+expectError(repoUrlFromPackage({}));
```

Comment:
I have been moving away from using `tsd` for projects with simple types. I realized that there really isn't much value in creating tests for simple types since they don't really catch anything you wouldn't already notice from reading the types.